### ISSUE TITLE
UnitTests.Semantics.BinaryOperators test failure on Fedora rawhide

### ISSUE THREAD

**Issue by omajid** [dotnet/roslyn#67101]
**Version Used**: 

https://github.com/dotnet/roslyn at commit 62a827b5880d9f935297f1a83964f92cd519a734

**Steps to Reproduce**:

1. Use a Fedora Rawhide container (`fedoraproject.org/fedora:rawhide`)
2. eng/build.sh --restore --build --pack
3. eng/build.sh --test

**Diagnostic Id**:

If this is a report about a bug in an analyzer, please include the diagnostic if possible (e.g. `"IDE0030"`).

**Expected Behavior**:

All tests pass

**Actual Behavior**:

```
Errors Microsoft.CodeAnalysis.VisualBasic.Semantic.UnitTests_15
[xUnit.net 00:04:27.58]     Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.BinaryOperators.Test5_DateConst [FAIL]
[xUnit.net 00:04:28.04]     Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.BinaryOperators.Test1_Date [FAIL]
...
Microsoft.CodeAnalysis.VisualBasic.Semantic.UnitTests_15                    FAILED 00:05:04.1415232 
```

**Comment by omajid** [comment]
It's very likely that this is an OS-specific issue to do with Fedora.

**Comment by omajid** [comment]
fyi @tmds

**Comment by tmds** [comment]
> It's very likely that this is an OS-specific issue to do with Fedora. Any idea how to verify that claim or how to rule it out?

If it fails on Fedora Rawhide and not on the stable Fedora version, that's a good indication.

The test failures are very different.

> Test5_DateConst

This one fails because of the output not being expected. If you look at the stacktrace, you can see the date format is expected to have a space, like: `8:30:00 AM`, and that space is not in the output: like: `8:30:00AM`.

This formatting may depend on something provided by the distro.

> Test1_Date

This test fails because it's unable to execute the assembly:

```
Execution failed for assembly 'VBBinaryOperators11, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
```

**Comment by tmds** [comment]
> The test failures are very different:

Actually, I looked at the Fedora test results, and the exception message for both is about the missing space in the time format.

**Comment by tmds** [comment]
> If you look at the stacktrace, you can see the time format is expected to have a space, like: 8:30:00 AM, and that space is not in the output: like: 8:30:00AM.

Taking a closer look.

```cs
TimeOnly time = new(23, 10, 59);

Console.WriteLine(time.ToString());
Console.WriteLine(time.ToString(CultureInfo.InvariantCulture));
```

If you run this on Fedora with `LANG=en_US.UTF-8`.

For Fedora 37, the output is:

```
11:10 PM
23:10:59.0000000
```

For Fedora Rawhide, the output is:

```
11:10PM
23:10:59.0000000
```

Both tests seem to be setting the locale to `en-US`.
I don't think even then the formatting is guaranteed to be the same on Windows and Linux.

The test fails on Rawhide because the locale-based formatting changed for time (`8:30:00 AM` vs `8:30:00AM`).
It may be intentional, or a regression.

**Comment by jcouv** [comment]
This doesn't sound like a compiler issue. @tmds Do you know how to route this?

**Comment by tmds** [comment]
After investigating further, I think this is a regression in Fedora.

I've reported a bug here: https://bugzilla.redhat.com/show_bug.cgi?id=2178563.

**Comment by tmds** [comment]
> After investigating further, I think this is a regression in Fedora.

And ... it is not a regression in Fedora.

The _SPACE_ in the format got replaced by a _NARROW NO-BREAK SPACE_.

So, we're in the case of:

> I don't think even then the formatting is guaranteed to be the same on Windows and Linux.

That means: either the test needs to be made culture invariant. Or it should be skipped on non-Windows.

It seems there's already a test being skipped in the file due to culture:

https://github.com/dotnet/roslyn/blob/65cd3dcc15df7fb39a138efc0faea84acd467816/src/Compilers/VisualBasic/Test/Semantic/Semantics/BinaryOperators.vb#L23-L24

@jcouv are you ok with skipping this test on non-Windows?

I can make a PR for it.

**Comment by tmds** [comment]
The plot thickens: .NET makes changes to what icu returns: https://github.com/dotnet/runtime/issues/83571.
It may be to compensate for differences between Windows and icu formatting.

**Comment by jcouv** [comment]
@tmds Yes, it would be okay to skip this test on non-Windows. In such cases, we mark the test with the link to Roslyn's follow-up issue.
